### PR TITLE
Removed the superfluous NetworkMapCacheImpl class

### PR DESCRIPTION
**Pull request by shamsasari**
NetworkMapCacheImpl and NetworkMapCacheBaseInternal interface merged into PersistentNetworkMapCache and NetworkMapCacheInternal interface respectively.

### CODE REVIEW

**Comment by shamsasari** [comment]
Addressed comments. I ended up removing the check on all the node-infos in `start`.

File: node/src/integration-test/kotlin/net/corda/node/services/messaging/ArtemisMessagingTest.kt [hunk 1]
```diff
@@ -80,8 +79,7 @@ class ArtemisMessagingTest {
         }
         LogHelper.setLevel(PersistentUniquenessProvider::class)
         database = configureDatabase(makeTestDataSourceProperties(), DatabaseConfig(), { null }, { null })
-        val persistentNetworkMapCache = PersistentNetworkMapCache(database, ALICE_NAME).apply { start(emptyList()) }
-        networkMapCache = NetworkMapCacheImpl(persistentNetworkMapCache, rigorousMock(), database).apply { start() }
+        networkMapCache = PersistentNetworkMapCache(database,  rigorousMock(), ALICE_NAME).apply { start(emptyList()) }
```
Comment (kasiastreich) [inline node/src/integration-test/kotlin/net/corda/node/services/messaging/ArtemisMessagingTest.kt]: Double space

File: node/src/main/kotlin/net/corda/node/services/network/PersistentNetworkMapCache.kt [hunk 2]
```diff
@@ -183,6 +197,27 @@ open class PersistentNetworkMapCache(private val database: CordaPersistence,
...
+            failures.forEach { logger.warn("", it) }
+            return false
+        }
+
+        for (identity in node.legalIdentitiesAndCerts) {
+            try {
+                identityService.verifyAndRegisterIdentity(identity)
+            } catch (ignore: Exception) {
+                // Log a warning to indicate node info is not added to the network map cache.
+                logger.warn("Node info for :'${identity.name}' is not added to the network map due to verification error.")
```
Comment (kasiastreich) [inline node/src/main/kotlin/net/corda/node/services/network/PersistentNetworkMapCache.kt]: It is added, you don't return false in this case

**Comment by shamsasari** [comment]
Back to you

File: node/src/main/kotlin/net/corda/node/services/network/PersistentNetworkMapCache.kt [hunk 3]
```diff
@@ -160,6 +173,7 @@ open class PersistentNetworkMapCache(private val database: CordaPersistence,
             val previousNode = getNodesByLegalIdentityKey(node.legalIdentities.first().owningKey).firstOrNull()
             if (previousNode == null) {
                 logger.info("No previous node found")
+                if (!verifyAndRegisterIdentities(node)) return
```
Comment (kasiastreich) [inline node/src/main/kotlin/net/corda/node/services/network/PersistentNetworkMapCache.kt]: When I think of it. We should do this check also on update.

Comment (shamsasari) [reply node/src/main/kotlin/net/corda/node/services/network/PersistentNetworkMapCache.kt]: Done, back to you.

**Pull request merged**